components/Pages/Grants/ExternalId/AddExternalIdDialog.tsx

```diff
@@ -0,0 +1,297 @@
+/* eslint-disable @next/next/no-img-element */
+import { FC, Fragment, ReactNode, useState } from "react";
+import { Dialog, Transition } from "@headlessui/react";
+import { PlusIcon, XMarkIcon } from "@heroicons/react/24/solid";
+import { z } from "zod";
+import { useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import { Button } from "@/components/Utilities/Button";
+import { MESSAGES } from "@/utilities/messages";
+
+import toast from "react-hot-toast";
+import fetchData from "@/utilities/fetchData";
+import { INDEXER } from "@/utilities/indexer";
+
+import { cn } from "@/utilities/tailwind";
+
+const inputStyle =
+  "bg-gray-100 border border-gray-400 rounded-md p-2 dark:bg-zinc-900";
+const labelStyle =
+  "text-slate-700 text-sm font-bold leading-tight dark:text-slate-200";
+
+const schema = z.object({
+  profile: z
+    .string()
+    .regex(
+      /^https:\/\/explorer\.gitcoin\.co\/#\/round\/(\d+)\/(\d+)\/(\d+)$/,
```

Comment:
if the user paste like `https://explorer.gitcoin.co/#/round/42161/25/83` it will fail

Reply:
It's not failing . The regex looks correct for me